### DISCUSSION TITLE
Action -> Execute Query in Sidepanel does not update the Table

### DISCUSSION THREAD

**Discussion by Markus-** [Budibase/budibase#10392] [Help] [answered]
Hi,

I have a screen with a details side panel where I want to update a row in a table. I had many problems by using the Save-Row-Action (duplicate key in many-many-SQL Table), so I created an update-Query which works as designed.

The external SQL-Table is updated and the side panel ist closed, but the Table on the original screen is not updated immediatly.

How do I tell the screen that the table changed?

Regards
Markus

**Comment by ConorWebb96** [comment]
Hey @Markus-,

When running a custom update query, you will need to refresh the tables data provider after the update has finished. To do this use the `Refresh Data Provider` action. (Screenshot below shows this) Make sure to select the Data provider you want to refresh. 
<img width="1079" alt="Screenshot 2023-04-24 at 11 23 19" src="https://user-images.githubusercontent.com/126772285/233969825-49e49522-37de-4dbc-aa6f-692873fda28f.png">

Additionally, make sure the edit form has access to the Data Provider which you wish to refresh. To do this drag the side panel inside the data provider where you are using the table. You should now be able to select the table's data provider. 

I hope this helps!

**Reply by ConorWebb96** [reply]
@Markus- would it be possible to get your sql schema/dump so I can look into the many to many issue you are experiencing?

Thanks in advance!

**Reply by Markus-** [reply]
Problem is the Dataprovider of the Table is not selectable in the Action of the Sidepanel.

**Reply by ConorWebb96** [reply, marked as answer]
I've attached a few screenshots to help explain what I meant earlier. 

This is before e.g. the pre-generated page using table blocks and then ejecting. (This is likely what you are currently seeing. It only shows the details data provider.)
<img width="1566" alt="Screenshot 2023-04-24 at 12 08 44" src="https://user-images.githubusercontent.com/126772285/233979852-239e0963-c0c7-40ed-9de1-bba3f35eaa9d.png">

This is what I'm suggesting. (I've moved the edit side panel into the data provider for the table so it now has access to the provider which needs refreshed.)
<img width="1560" alt="Screenshot 2023-04-24 at 12 09 13" src="https://user-images.githubusercontent.com/126772285/233979857-d04ad860-223a-4ce4-8f11-0b70b7c245f4.png">

I hope the above makes sense and is helpful!

**Comment by Markus-** [comment]
It is a complex schema.

In short:
- Products -> Services (many-to-many) through a stock-Table
- Products -> Approvers (many-to-many) through a product approver Table
- Products -> Roles (one-to-many) 

When I add an auto-generated screen for the product-Table it simply does not work when updating approvers or services. I'll get a duplicated key error and the postgres-log shows inserting problems with the stock table. I don't get why there is an insert despite I update something or in case of an updated approver the stocktable must not be handled...

**Reply by Markus-** [reply]
I will open a separate discussion.
